packages/abi-utils/lib/parse.ts

```diff
@@ -0,0 +1,138 @@
+import { Parameter, FunctionEntry, ErrorEntry, EventEntry } from "./types";
+
+export function parseFunctionSignature(signature: string): FunctionEntry {
+  const { name, inputs } = parseSignature(signature);
+  return {
+    type: "function" as const,
+    name,
+    inputs,
+    outputs: [] as Parameter[],
+    stateMutability: "nonpayable" as const
+  };
+}
+
+export function parseErrorSignature(signature: string): ErrorEntry {
+  const { name, inputs } = parseSignature(signature);
+  return {
+    type: "error" as const,
+    name,
+    inputs
+  };
+}
+
+export function parseEventSignature(signature: string): EventEntry {
+  const { name, inputs } = parseSignature(signature);
+  return {
+    type: "event" as const,
+    name,
+    inputs: inputs.map(parameter => ({ ...parameter, indexed: false })),
+    anonymous: false
+  };
+}
+
+export function parseSignature(signature: string): {
+  name: string;
+  inputs: Parameter[];
+} {
+  let openParenIndex = signature.indexOf("(");
+  if (openParenIndex === -1) {
+    openParenIndex = signature.length; //set to end of string if not found
+  }
+  const parameterList = signature.slice(openParenIndex);
+  const name = signature.slice(0, openParenIndex);
+  const inputs = parseParameterList(parameterList);
+  return {
+    name,
+    inputs
+  };
+}
+
+export function parseParameterList(parameterList: string): Parameter[] {
```

Comment:
Does this (and the other parameter list one) need to be exported? Or are these more internal concerns?

Reply:
Removed export on parse-with-remainder; left it on on the others.